### PR TITLE
Nits

### PR DESCRIPTION
**Pull request by kolyshkin**
Lots of small changes, needs a careful review...

### 1. pkg/cgroups/createCgroupv2Path: nits
    
1. Check the path validity before trying to read the cgroup.controllers.  
2. Do not hardcode "/sys/fs/cgroup".
3. Simplify creating the "+this +that" string.
4. Do not wrap ioutil.WriteFile error.

### 2. Remove excessive error wrapping

In case os.Open[File], os.Mkdir[All], ioutil.ReadFile and the like
fails, the error message already contains the file name and the
operation that fails, so there is no need to wrap the error with
something like "open %s failed".
    
While at it
    
- replace a few places with os.Open, ioutil.ReadAll with
ioutil.ReadFile.
    
- replace errors.Wrapf with errors.Wrap for cases where there
are no %-style arguments.

### CODE REVIEW

File: libpod/container_internal_linux.go [hunk 1]
```diff
@@ -835,13 +835,13 @@ func (c *Container) checkpointRestoreLabelLog(fileName string) error {
 
 	logFile, err := os.OpenFile(dumpLog, os.O_CREATE, 0600)
 	if err != nil {
-		return errors.Wrapf(err, "failed to create CRIU log file %q", dumpLog)
+		return err
```
Comment (mheon) [inline libpod/container_internal_linux.go:838]: I think the CRIU log file bit of the error might be useful here.

File: libpod/container_internal_linux.go [hunk 2]
```diff
@@ -1125,16 +1125,10 @@ func (c *Container) restore(ctx context.Context, options ContainerCheckpointOpti
...
-				return errors.Wrapf(err, "Failed to open deleted files file %s", deletedFilesPath)
-			}
-			defer deletedFilesFile.Close()
-
 			var deletedFiles []string
-			deletedFilesJSON, err := ioutil.ReadAll(deletedFilesFile)
+			deletedFilesJSON, err := ioutil.ReadFile(deletedFilesPath)
 			if err != nil {
-				return errors.Wrapf(err, "Failed to read deleted files file %s", deletedFilesPath)
+				return errors.Wrapf(err, "Failed to read deleted files file")
```
Comment (mheon) [inline libpod/container_internal_linux.go:1131]: F in Failed needs to be lowercased... I'm actually surprised Lint didn't catch this before your PR.

**Comment by mheon** [comment]
@giuseppe PTAL at the `pkg/cgroups` bit please

**Comment by mheon** [comment]
Few nits, but LGTM overall

File: libpod/oci_conmon_linux.go [hunk 3]
```diff
@@ -157,15 +157,13 @@ func newConmonOCIRuntime(name string, paths []string, conmonPath string, runtime
...
-				runtime.exitsDir)
+			return nil, errors.Wrapf(err, "error creating OCI runtime exit files directory")
 		}
 	}
 	if err := os.MkdirAll(runtime.socketsDir, 0750); err != nil {
 		// The directory is allowed to exist
 		if !os.IsExist(err) {
-			return nil, errors.Wrapf(err, "error creating OCI runtime attach sockets directory %s",
-				runtime.socketsDir)
+			return nil, errors.Wrap(err, "error creating OCI runtime attach sockets directory")
```
Comment (TomSweeneyRedHat) [inline libpod/oci_conmon_linux.go:166]: For these directory errors, will the directory that can't be created be included in the error message somewhere else?  I think that's important to include in the message.

Comment (rhatdan) [reply libpod/oci_conmon_linux.go:166]: The err returned my os.MkdirAll will include the path.

**Review by rhatdan — APPROVED** [review]
LGTM other then @mheon issue.

File: libpod/container_internal_linux.go (hunk 1 above)
Comment (kolyshkin) [reply libpod/container_internal_linux.go:838]: fixed

File: libpod/container_internal_linux.go (hunk 2 above)
Comment (kolyshkin) [reply libpod/container_internal_linux.go:1131]: Will address by a separate commit (in this PR).

Comment (kolyshkin) [reply libpod/container_internal_linux.go:1131]: There are other 100+ places like this :)

See the new commit

Comment (kolyshkin) [reply libpod/container_internal_linux.go]: Some of the remaining cases can be seen by running
```console
[kir@kir-rhat libpod]$ git grep -F errors. | grep -v ^vendor | grep '"[A-Z]'
```

but I don't have the courage to fix.

Comment (mheon) [reply libpod/container_internal_linux.go]: Well that's awfully concerning - I thought we had a linter to catch this, but I'm betting we lost it in the move to golangci-lint.

**Comment by giuseppe** [comment]
LGTM

**Review by giuseppe — APPROVED** [review]

**Review by vrothberg — COMMENTED** [review]
LGTM, thanks @kolyshkin !

**Pull request merged**